R/dbi-connection.R

```diff
@@ -191,6 +191,11 @@ setMethod("dbListTables", "OdbcConnection",
            table_name = NULL,
            table_type = NULL,
            ...) {
+    check_string(catalog_name, allow_na = TRUE, allow_null = TRUE)
```

Comment:
Generally, w.r.t. `allow_na`, I tried to err on the side of letting things slip by that I ought not to rather than than mistakenly erroring for input that's okay.

Reply:
Hmmm, I think I'd tend to trend in the opposite direction, especially for these functions where we're explicitly using `NULL` as the "default" option.